### PR TITLE
Bug/1867 analytics report order

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #1867

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: includes/Modules/Analytics.php [hunk 1]
```diff
@@ -1094,6 +1062,48 @@ function ( $order_def ) {
...
+						array(
+							'fieldName' => '',
+							'sortOrder' => 'DESCENDING',
+						),
+						(array) $order_def
+					);
+
+					if ( empty( $order_def['fieldName'] ) ) {
+						return null;
+					}
```
Comment (aaemnnosttv) [inline includes/Modules/Analytics.php:1093]: I know @felixarntz suggested adding the default `sortOrder` back in the previous PR but if we're going to filter out invalid orderbys then I think we should require both to be provided rather than just the `fieldName` and leave out any defaults. The merging there is mainly to ensure the structure to prevent non-existing key warnings, but also because orderbys used to be actual objects due to the way batch requests worked previously.
I'm happy to defer to @felixarntz on this, but in my opinion we should avoid any default here and require both or filter it out.

File: tests/phpunit/integration/Modules/AnalyticsTest.php [hunk 2]
```diff
@@ -458,4 +461,50 @@ protected function get_module_with_screen() {
...
+		$order  = array(
+			'fieldName' => 'views',
+			'sortOrder' => 'ASCENDING',
+		);
+		$result = $reflected_parse_data_orderby_method->invoke( $analytics, new Data_Request( 'GET', null, null, 'report', array( 'orderby' => $order ) ) );
+		$this->assertTrue( is_array( $result ) );
+		$this->assertEquals( 1, count( $result ) );
+		$this->assertTrue( $result[0] instanceof Google_Service_AnalyticsReporting_OrderBy );
+		$this->assertEquals( $order['fieldName'], $result[0]->getFieldName() );
+		$this->assertEquals( $order['sortOrder'], $result[0]->getSortOrder() );
```
Comment (aaemnnosttv) [inline tests/phpunit/integration/Modules/AnalyticsTest.php:486]: Please use the string literals for the expected values rather than referencing the `$order` variable. They should be the same, but this makes the test easier to read and less likely to pass when it shouldn't if something is wrong in the input.

File: tests/phpunit/integration/Modules/AnalyticsTest.php [hunk 3]
```diff
@@ -458,4 +461,50 @@ protected function get_module_with_screen() {
...
+		$result = $reflected_parse_data_orderby_method->invoke( $analytics, new Data_Request( 'GET', null, null, 'report', array( 'orderby' => $orders ) ) );
+		$this->assertTrue( is_array( $result ) );
+		$this->assertEquals( 2, count( $result ) );
+		$this->assertTrue( $result[0] instanceof Google_Service_AnalyticsReporting_OrderBy );
+		$this->assertEquals( $orders[0]['fieldName'], $result[0]->getFieldName() );
+		$this->assertEquals( $orders[0]['sortOrder'], $result[0]->getSortOrder() );
+		$this->assertTrue( $result[1] instanceof Google_Service_AnalyticsReporting_OrderBy );
+		$this->assertEquals( $orders[1]['fieldName'], $result[1]->getFieldName() );
+		$this->assertEquals( $orders[1]['sortOrder'], $result[1]->getSortOrder() );
+	}
```
Comment (aaemnnosttv) [inline tests/phpunit/integration/Modules/AnalyticsTest.php:508]: Let's also test that an orderby is not returned if it's `fieldName` is empty. Pending my previous suggestion, we should do the same if the `sortOrder` is also empty (currently this would never happen though since it has the default).

File: tests/phpunit/integration/Modules/AnalyticsTest.php (hunk 2 above)
Comment (eugene-manuilov) [reply tests/phpunit/integration/Modules/AnalyticsTest.php]: Updated.

File: tests/phpunit/integration/Modules/AnalyticsTest.php (hunk 3 above)
Comment (eugene-manuilov) [reply tests/phpunit/integration/Modules/AnalyticsTest.php:532]: I have added a test to check the case when `fieldName` is empty. Will udpate it if Felix agrees with your suggestion.

File: includes/Modules/Analytics.php (hunk 1 above)
Comment (felixarntz) [reply includes/Modules/Analytics.php]: SGTM!

Comment (eugene-manuilov) [reply includes/Modules/Analytics.php]: Updated.

File: tests/phpunit/integration/Modules/AnalyticsTest.php (hunk 3 above)
Comment (eugene-manuilov) [reply tests/phpunit/integration/Modules/AnalyticsTest.php:536]: Updated.

File: includes/Modules/Analytics.php [hunk 4]
```diff
@@ -1094,6 +1062,50 @@ function ( $order_def ) {
 		throw new Invalid_Datapoint_Exception();
 	}
 
+	/**
+	 * Parses orderby value of the data request.
```
Comment (aaemnnosttv) [inline includes/Modules/Analytics.php:1066]:
```suggestion
	 * Parses the orderby value of the data request into an array of reporting orderby object instances.
```

File: includes/Modules/Analytics.php [hunk 5]
```diff
@@ -1094,6 +1062,50 @@ function ( $order_def ) {
...
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $orderby Data request object.
```
Comment (aaemnnosttv) [inline includes/Modules/Analytics.php:1070]:
```suggestion
	 * @param array|null $orderby Data request orderby value.
```

File: includes/Modules/Analytics.php [hunk 6]
```diff
@@ -1094,6 +1062,50 @@ function ( $order_def ) {
 		throw new Invalid_Datapoint_Exception();
 	}
 
+	/**
+	 * Parses the orderby value of the data request into an array of reporting orderby object instances.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array|null $orderby Data request orderby value.
+	 *
```
Comment (felixarntz) [inline includes/Modules/Analytics.php:1071]:
```suggestion
```
